.travis.yml

```diff
@@ -29,50 +29,49 @@ cache:
 
 jobs:
   include:
-    - stage: "Build front and upload docker image"
-      before_script:
+    - stage: "Build the frontend and upload a Docker image"
+      script:
         - cd front
         - npm install
         - npm run build
         - cd -
-      script:
-        - export BRANCH="$(echo "$TRAVIS_BRANCH" | tr '/' '-')"
-        - make front/travis-docker "tag=$BRANCH--front"
-        - docker login quay.io -u "$DOCKER_USER" -p "$DOCKER_PASS"
-        - docker push "quay.io/aelve/guide:$BRANCH--front"
-      if: type = push
+        - |
+          if [ "$TRAVIS_EVENT_TYPE" = "push" ]; then
+            export BRANCH="$(echo "$TRAVIS_BRANCH" | tr '/' '-')"
+            make front/travis-docker "tag=$BRANCH--front"
+            docker login quay.io -u "$DOCKER_USER" -p "$DOCKER_PASS"
+            docker push "quay.io/aelve/guide:$BRANCH--front"
+          fi
 
-    - stage: "Build backend dependencies"
```

Comment:
I consolidated the stages because each stage brought about 3 minutes of CI time overhead (loading the cache, unloading the cache).